### ISSUE TITLE
cli: add support for specifying container options for each step of a workflow 

### ISSUE THREAD

**Issue by Jswig** [getpopper/popper#876]
Add a way of providing container configuration options for each step, e.g.
```yaml
steps:
- id: foo
  uses: ./
  options:
    volumes:
    - /my/new/volume:/inside
    ports:
    - 8888: 80
```
For instance, it will be useful for:
-  enabling the use options with `popper sh`, which is not currently possible and will make it easier for researchers to develop workflows interactively
- developing workflows where steps require different/ conflicting options, e.g. different volumes, port mappings

**Comment by Jswig** [comment]
@ivotron 
Which options should have priority when we pass a configuration file to `popper run`?

**Comment by ivotron** [comment]
we can have step have higher precedence than the options coming from a config file

**Comment by ivotron** [comment]
closed in #881
